zcash_primitives/CHANGELOG.md

```diff
@@ -6,11 +6,22 @@ and this library adheres to Rust's notion of
 [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- `zcash_client_backend::AccountBalance::with_unshielded_balance_mut`
 
+### Deprecated
+- `zcash_client_backend::AccountBalance::add_unshielded_value`
+- `zcash_client_backend::AccountBalance::unshielded`
```

Comment:
`zcash_client_backend::AccountBalance::unshielded` is removed, not deprecated.

Option 1 from above:
```suggestion
### Removed
- `zcash_client_backend::AccountBalance::unshielded`. Instead use
  `account_balance.unshielded_balance().total()`.

### Deprecated
- `zcash_client_backend::AccountBalance::add_unshielded_value`. Instead use
  `AccountBalance::with_unshielded_balance_mut` with a closure that calls
  the appropriate `add_*_value` method(s) of `Balance` on its argument.
  Note that the appropriate method(s) depend on whether the funds are
  spendable, pending change, or pending non-change (previously, only the
  total unshielded value was tracked). The deprecated implementation now
  assumes that they are pending non-change, which may not be correct, and
  so all uses of this method should be checked and replaced.
```

Option 2 from above:
```suggestion
### Removed
- `zcash_client_backend::AccountBalance::unshielded`. Instead use
  `account_balance.unshielded_balance().total()`.
- `zcash_client_backend::AccountBalance::add_unshielded_value`. Instead use
  `AccountBalance::with_unshielded_balance_mut` with a closure that calls
  the appropriate `add_*_value` method(s) of `Balance` on its argument.
  Note that the appropriate method(s) depend on whether the funds are
  spendable, pending change, or pending non-change (previously, only the
  total unshielded value was tracked).
```

Reply:
I'm settling for option two (if possible) since it would be a better contribution overall